packages/@aws-cdk/aws-synthetics/README.md

```diff
@@ -1,16 +1,206 @@
+
+> The APIs of higher level constructs in this module are experimental and under active development. They are subject to non-backward compatible changes or removal in any future version. These are not subject to the [Semantic Versioning](https://semver.org/) model and breaking changes will be announced in the release notes. This means that while you may use them, you may need to update your source code when upgrading to a newer version of this package.
+
 ---
 <!--END STABILITY BANNER-->
 
-This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
+Amazon CloudWatch Synthetics allow you to monitor your application by generating **synthetic** traffic. The traffic is produced by a **canary**. A canary is a configurable script that runs on a schedule. You configure the canary script to follow the same routes and perform the same actions as a customer, which allows you to continually verify your customer experience even when you don't have any customer traffic on your applications.
+
+## Canary
+
+To illustrate how to use a canary, assume your application defines the following endpoint:
+
+```bash
+curl https://api.awsomesite.com/user/books/topbook/
+The Hitchhikers Guide to the Galaxy
+
+```
+
+The below code defines a canary that will hit the `books/topbook` endpoint every 5 minutes:
+
+> **Not Implemented:** the `test` property will be a part of Milestone 2.
+
+```ts
+const canary = new Canary(this, 'MyCanary', {
+  name: 'mycanary'
+  frequency: Duration.minutes(5),
+  test: synth.Test.custom({
+    code: synth.Code.fromInline(`const https = require('https');
+      var synthetics = require('Synthetics');
+      const log = require('SyntheticsLogger');
+  
+      exports.handler = async function () {
+        const requestOptions = {"hostname":"api.awsomesite.com","method":"","path":"/user/books/topbook/","port":443}
+        let req = https.request(requestOptions);
+        req.on('response', (res) => {
+          log.info()
+        });
+      }`),
+    handler: 'index.handler',
+  }),
+});
+```
+
+The canary will automatically produce a CloudWatch Dashboard:
+
+![UI Screenshot](images/ui-screenshot.png)
+
+### Canary Test Property
+
+The `test` property represents the test that the canary executes. It accepts a type `Test`, which has as parameters the properties required to generate the test script.
+
+> **Not Implemented:** `Test.heartBeat()` is a part of Milestone 3
+
+Testing URL heartbeat:
+
+```ts
+const canary = new Canary(this, 'MyCanary', {
+  test: Test.heartBeat('https://myawsomesite.com'),
+  // ...
+});
+```
+
+> **Not Implemented:** `Test.apiEndpoint()` is a part of Milestone 4
+
+Testing a specific API:
+
+```ts
+const canary = new Canary(this, 'MyCanary', {
+  test: Test.apiEndpoint('https://myawsomesite.com/endpoint', {
+    method: 'GET'|'POST',
+    headers: Map<headerName, headerValue>,
+    data: string,
```

Comment:
What is this even? It's not valid TypeScript (type signatures used as an object literal??)

Reply:
Am removing this for now but when I implement this API I will add in a valid TS example :).